framework/platform/bpl/uci/db/bpl_db_uci.h

```diff
@@ -0,0 +1,134 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _BPL_DB_UCI_H_
+#define _BPL_DB_UCI_H_
+
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace beerocks {
+namespace bpl {
+namespace db {
+
+/****************************************************************************/
+/******************************* Definitions ********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************* Structures *********************************/
+/****************************************************************************/
+
+/****************************************************************************/
+/******************************** Functions *********************************/
+/****************************************************************************/
+
+/**
+ * @brief Find if entry exists.
+ * 
+ * @param[in] config_file name of the requested configuration.
+ * @param[in] entry_type type of the requested entry.
+ * @param[in] entry_name name of the requested entry.
+ * @return true if entry exists in db, false otherwise
+ */
+bool uci_entry_exists(const std::string &config_file, const std::string &entry_type,
+                      const std::string &entry_name);
```

Comment:
uci.h itself has a uci_context which has to be passed in each call. Therefore, it would make sense to put all this in a class that carries the uci_context (and that will take care of loading and saving).

Also, as @mariomaz will tell you, this class should be split into two classes: an interface class that only has pure virtual methods, and an implementation class that overrides them. That will make sure you can write a unit test for the peristent DB without having an actual UCI.

Reply:
While I agree with your suggestion to create a class instance to handle the UCI interactions, we wanted to provide a quick solution to be used for the persistent DB feature and we plan to later-on refactor the BPL UCI interface.

https://jira.prplfoundation.org/browse/PPM-315